### ISSUE TITLE
Driver should check the data format of Payload

### ISSUE THREAD

**Issue by AhaEdgar** [edgexfoundry/device-mqtt-go#9]
https://github.com/edgexfoundry/device-mqtt-go/blob/e415b6d3d289a5950d118ac1f5b2a1022bcd2a3c/internal/driver/incominglistener.go#L59

```
func onIncomingDataReceived(client mqtt.Client, message mqtt.Message) {
	var response map[string]interface{}
	json.Unmarshal(message.Payload(), &response)

	// add check the data format by Geoffrey
	if response == nil || response["name"] == nil || response["cmd"] == nil {
		driver.Logger.Warn(fmt.Sprintf("[Incoming listener] Incoming reading ignored. Data Format Error, couldn't be converted to Map[String]interface{}  : topic=%v payload=%v", message.Topic(), string(message.Payload())))
		return
	}

	deviceName := response["name"].(string)
	cmd := response["cmd"].(string)
	reading := response[cmd]
```

Action： add check the data format of Payload by Geoffrey

Reason：
      If the Payload is  `"testmest"` or `{"run":"100"}···, it will make the `device-mqtt-go` service exit by the panic `panic: interface conversion: interface {} is nil, not string`. Because these non-standard formats cannot be converted to `map[string]interface{}`. Even though the Payload could be converted to `map[string]interface{}`, it must hava map["name"] and map["cmd"]. If not, the  the `device-mqtt-go` exit by the panic `panic: interface conversion: interface {} is nil, not string`. 
     As a DeviceService, the wrong data type can be left unprocessed, but should not cause the service to be unavailable.
      Therefore, we should verify the data format of Payload to prevent the `device-mqtt-go` from exiting abnormally due to data format problems. In addition, the availability of the `device-mqtt-go` is guaranteed. Maybe the way I deal with is not elegant enough, but I hope to attract attention to this kind of problem.

By Geoffrey

**Comment by weichou1229** [comment]
fix by #11

**Comment by cloudxxx8** [comment]
@AhaEdgar 
We have fixed this issue in #11
Could you please help try again and see whether we can close this issue?
